### PR TITLE
fix(cli): proxy support is broken

### LINKED ISSUE #5743: CDK 1.20 Unable to determine the default AWS account
CDK v1.20 does not set `CDK_DEFAULT_ACCOUNT` when `AWS_SESSION_TOKEN`, 
 `AWS_SECRET_ACCESS_KEY`, and `AWS_ACCESS_KEY_ID` are set in the environment. Works fine in v1.19. 

I cannot use a CLI profile because my organisation requires MFA authentication; as work around I am starting a session via `boto3` and exporting the environment variables.

### Reproduction Steps

```
$ printenv |grep -e AWS -e CDK
AWS_SESSION_TOKEN=Fxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxo=
AWS_DEFAULT_REGION=eu-west-2
AWS_SECRET_ACCESS_KEY=sxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx8
AWS_ACCESS_KEY_ID=Axxxxxxxxxxxxxxxxxx7

$ ls ~/.aws   # deliberately no profile
ls: cannot access /home/xxxxx/.aws: No such file or directory

$ npm install -g aws-cdk@1.19
$ rm -fr ~/.cdk
$ cdk synth -v 2>v1.19.stderr
$ npm install  -g aws-cdk@1.20
$ rm -fr ~/.cdk
$ cdk synth -v 2>v1.20.stderr

$ diff v1.19.stderr v1.20.stderr
1c1
< CDK toolkit version: 1.19.0 (build 5597bbe)
---
> CDK toolkit version: 1.20.0 (build 021c521)
26a27
> Does not look like EC2 instance.
30d30
< Does not look like EC2 instance.
43,44c43,60
< Default account ID: 3xxxxxxxxxx2
< Setting "CDK_DEFAULT_ACCOUNT" environment variable to 3xxxxxxxxxx2
---
> Unable to determine the default AWS account (did you configure "aws configure"?): { 503: 
>     at Request.extractError (/xxxxxxxxxxxxxx/.nvm/versions/node/v10.16.3/lib/node_modules/aws-cdk/node_modules/aws-sdk/lib/protocol/query.js:55:29)
>     at Request.callListeners (/xxxxxxxxxxxxxx/.nvm/versions/node/v10.16.3/lib/node_modules/aws-cdk/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
>     at Request.emit (/xxxxxxxxxxxxxx/.nvm/versions/node/v10.16.3/lib/node_modules/aws-cdk/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
>     at Request.emit (/xxxxxxxxxxxxxx/.nvm/versions/node/v10.16.3/lib/node_modules/aws-cdk/node_modules/aws-sdk/lib/request.js:683:14)
>     at Request.transition (/xxxxxxxxxxxxxx/.nvm/versions/node/v10.16.3/lib/node_modules/aws-cdk/node_modules/aws-sdk/lib/request.js:22:10)
>     at AcceptorStateMachine.runTo (/xxxxxxxxxxxxxx/.nvm/versions/node/v10.16.3/lib/node_modules/aws-cdk/node_modules/aws-sdk/lib/state_machine.js:14:12)
>     at /xxxxxxxxxxxxxx/.nvm/versions/node/v10.16.3/lib/node_modules/aws-cdk/node_modules/aws-sdk/lib/state_machine.js:26:10
>     at Request.<anonymous> (/xxxxxxxxxxxxxx/.nvm/versions/node/v10.16.3/lib/node_modules/aws-cdk/node_modules/aws-sdk/lib/request.js:38:9)
>     at Request.<anonymous> (/xxxxxxxxxxxxxx/.nvm/versions/node/v10.16.3/lib/node_modules/aws-cdk/node_modules/aws-sdk/lib/request.js:685:12)
>     at Request.callListeners (/xxxxxxxxxxxxxx/.nvm/versions/node/v10.16.3/lib/node_modules/aws-cdk/node_modules/aws-sdk/lib/sequential_executor.js:116:18)
>   message: null,
>   code: 503,
>   time: 2020-01-10T11:23:24.090Z,
>   requestId: undefined,
>   statusCode: 503,
>   retryable: true }
> Setting "CDK_DEFAULT_ACCOUNT" environment variable to undefined
49c65
<   CDK_DEFAULT_ACCOUNT: '3xxxxxxxxxx2',
---
>   CDK_DEFAULT_ACCOUNT: undefined,
54c70
<   CDK_CLI_VERSION: '1.19.0' }
---
>   CDK_CLI_VERSION: '1.20.0' }

```
### Error Log

Unable to determine the default AWS account 

### Environment

  - **CLI Version      : 1.20.0**
  - **Framework Version: 1.20.0**
  - **OS               : Linux**
  - **Language         : Python**
--- 

This is :bug: Bug Report

### LINKED ISSUE #5791: CDK Synth doesn't create CloudFormation without full credentials (1.20.0)
In CDK version 1.20.0 the command CDK synth now needs full credentials to run. This is an undocumented (?) breaking change. Creating a CloudFormation template for an account and region without the linked secret key and key ID is no longer possible. This means that my company currently cannot use 1.20.0, as the access key id and secret access key would not be available in the development of the templates. For automatic testing this would also be breaking, as we would obviously not want to have the full credentials in the repository that would do an integration test. Surely CDK should (continue to) be able to create templates without needing deployment credentials at that stage. 

### Reproduction Steps

Create an environment with an AWS_REGION and an AWS_ACCOUNT, but without a matching AWS_ACCESS_KEY_ID or AWS_SECRET_ACCESS_KEY.

### Error Log

[Error at /test/skeleton] Need to perform AWS calls for account [redacted], but no credentials found. Tried: default credentials.

### Environment

  - **CLI Version      :1.20.0
  - **Framework Version:1.20.0
  - **OS               :OS Mac Catalina
  - **Language         :Python 3.6.5

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by rix0rrr**
Proxy support was broken in the PR that introduced support for custom CA
bundles. Fix the support.

Fixes #5743, fixes #5791.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk/lib/api/util/sdk.ts [hunk 1]
```diff
@@ -207,19 +207,29 @@ export class SDK implements ISDK {
...
-    options.caBundlePath = options.caBundlePath || caBundlePathFromEnvironment();
+    const proxyAddress = options.proxyAddress || httpsProxyFromEnvironment();
+    const caBundlePath = options.caBundlePath || caBundlePathFromEnvironment();
 
-    if (options.proxyAddress) { // Ignore empty string on purpose
-      debug('Using proxy server: %s', options.proxyAddress);
-      httpOptions.proxy = options.proxyAddress;
+    if (proxyAddress && caBundlePath) {
+      throw new Error(`Cannot specify Proxy (${proxyAddress}) and CA Bundle (${caBundlePath}) at the same time`);
+      // Maybe it's possible after all, but I've been staring at
```
Comment (eladb) [inline packages/aws-cdk/lib/api/util/sdk.ts]: If I understand correctly, there is a [use case](https://github.com/aws/aws-cdk/issues/5743#issuecomment-574059497) for using a proxy and a CA bundle, no?

Comment (rix0rrr) [reply packages/aws-cdk/lib/api/util/sdk.ts]: Probably. 

1) We didn't have CA certs before so not adding the feature is not a regression from that respect.

2) This PR is fixing the regression of proxy support working at all. Adding support for CAs while using the ProxyAgent can be a feature request that we deal with a later date. Have you seen the source code linked? I really don't want to add more risk by trying to figure out what is happening there that at this time.

Comment (eladb) [reply packages/aws-cdk/lib/api/util/sdk.ts]: Fair enough. so maybe we can improve the error message to read:

```
At the moment, it is impossible to use a proxy with a custom CA bundle (see issue https://github...).
```

**Comment by RenkeMeuwese** [comment]
Behavior in #5791 is unchanged with this version.

**Pull request merged**